source/types/trinary.cpp

```diff
@@ -39,9 +39,22 @@ static std::vector<std::vector<int8_t>> trytesTrits = {
   { 0, -1, 0 }, { 1, -1, 0 }, { -1, 0, 0 }
 };
 
+size_t
+tryteIndex(const char& tryte) {
+  if (tryte == '9') {
+    return 0;
+  }
+  // explicitly excluding '@' here, since by the calculation
+  // below this would map to 0, but 0 is the index for '9'
+  if (tryte == '@') {
+    return -1;
+  }
+  return tryte - 'A' + 1;
```

Comment:
I would prefer something consistent: return >= 0 for valid tryte
return -1 for non valid tryte.
The check for @ is quite confusing IMO, I'd prefer something like if >= 'A' && <= 'Z' then - 'A' + 1, otheriwse -1

Reply:
Thanks for the review. I adjusted it according to your suggestions and I agree, its much better now!